### PR TITLE
Fixes being unable to hit trophy cases in melee

### PR DESCRIPTION
**Pull request by Dorsisdwarf**
This was somewhat annoying

:cl: 
fix: Fixed being unable to hit museum cases in melee
/:cl:

(First PR in eleven trillion years watch me fuck it up somehow)

### CODE REVIEW

**Comment by Profakos** [comment]
Probably needs an openable check on line 144 (on which I can't comment somehow).

I shouldn't have fixed this in my PR, but I really didn't want to touch the attackby of the parent proc, but now I see that this is needed.

**Comment by RemieRichards** [comment]
@Profakos 144 is inside the if so it's already covered for ```openable```

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -13,6 +13,7 @@
 	var/obj/item/showpiece = null
 	var/alert = 0
 	var/open = 0
+	var/openable = 1
```
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm]: Use TRUE/FALSE defines please. Also if you could make the old variables use them please.

**Comment by Profakos** [comment]
No, it is inside the else if in 135 which is entered if ` if(W.GetID() && !broken && openable)` evaluates to false, and we are using the crowbar, so it can be unopenable, but it will still let us crowbar it open. But I could be seeing the indentation incorrectly.

**Comment by RemieRichards** [comment]
oh yeah right, github didn't expand that part of the diff.

**Review by AnturK — COMMENTED** [review]
\+ what Profakos said, or alternatively override toggle_lock to do something trophy specific.

**Comment by Incoming5643** [comment]
will conflict with #26869

File: code/game/objects/structures/displaycase.dm (hunk 1 above)
Comment (Dorsisdwarf) [inline code/game/objects/structures/displaycase.dm]: I dont know what an old variable is

Comment (Dorsisdwarf) [inline code/game/objects/structures/displaycase.dm]: Oh, I see

File: code/game/objects/structures/displaycase.dm [hunk 2]
```diff
@@ -278,6 +280,9 @@
 
 	if(!user.Adjacent(src)) //no TK museology
 		return
+	if(user.a_intent == INTENT_HARM)
+		..()
+		return
```
Comment (lzimann) [inline code/game/objects/structures/displaycase.dm]: Do a `return ..()` here

**Pull request merged**